core/templates/pages/story-editor-page/story-editor-page.controller.ts

```diff
@@ -59,19 +61,16 @@ angular.module('oppia').directive('storyEditorPage', [
           var TOPIC_EDITOR_URL_TEMPLATE = '/topic_editor/<topicId>';
           ctrl.returnToTopicEditorPage = function() {
             if (UndoRedoService.getChangeCount() > 0) {
-              var modalInstance = $uibModal.open({
+              $uibModal.open({
                 templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
                   '/pages/story-editor-page/modal-templates/' +
                   'story-save-pending-changes-modal.template.html'),
                 backdrop: true,
-                controller: [
-                  '$scope', '$uibModalInstance',
-                  function($scope, $uibModalInstance) {
-                    $scope.cancel = function() {
-                      $uibModalInstance.dismiss('cancel');
-                    };
-                  }
-                ]
+                controller: 'ConfirmOrCancelModalController'
+              }).result.then(function() {}, function() {
```

Comment:
If aren't doing any operation after the modal has closed, can we get rid of the then part here? Does it error without it?

Reply:
We need to use then here, otherwise will thrown an error because we are don't "handling" possible errors when catch is called.

Reply:
@kevintab95